presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1143,13 +1142,6 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
 
         List<PartitionUpdate> partitionUpdates = getPartitionUpdates(fragments);
 
-        if (isWritingStagingFilesEnabled(session)) {
```

Comment:
Should we also do the migration in this commit to use the `PageSinkProperties#isPartitionCommitRequired()`?

Reply:
Let's keep them separate to make the logic looks like "remove old session property -> use new property".

Reply:
Now I think about it, I feel it is not consistent in a sense that after removing the session property, we removed staging file writing and committing altogether. If we migrate the usage here, it means we only removed staging file committing (this is only included in recoverable grouped execution).

Reply:
Per offline discussion, staging file commit is now a different API in `Metadata` (`commitLifespanForCreate` and `commitLifespanForInsert`, yeah we could replace `lifespan` with `partition` as well), which is only used when recoverable execution is introduced. 

We don't want to introduce `PageSinkProperties` without actual usage, so we could check this field always be false in a separate PR and put the staging file writing code inside.